core/local/steps/incomplete_fixer.js

```diff
@@ -150,25 +154,30 @@ function step (incompletes /*: IncompleteItem[] */, opts /*: IncompleteFixerOpti
         }
 
         try {
+          let rebuilt
           if (wasRenamedSuccessively(item, event)) {
             // We have a match, try to rebuild the incomplete event
-            const rebuilt = await rebuildIncompleteEvent(item, event, opts)
-            log.debug({path: rebuilt.path, action: rebuilt.action}, 'rebuilt event')
-            if (rebuilt.path === event.path) {
-              batch.splice(batch.indexOf(event), 1, rebuilt)
-            } else {
-              batch.push(rebuilt)
-            }
+            rebuilt = await rebuildIncompleteEvent(item, event, opts)
           } else if (itemDestinationWasDeleted(item, event)) {
             // We have a match, try to replace the incomplete event
-            const rebuilt = buildDeletedFromRenamed(item, event)
-            log.debug({path: rebuilt.path, action: rebuilt.action}, 'rebuilt event')
-            batch.push(rebuilt)
+            rebuilt = buildDeletedFromRenamed(item, event)
```

Comment:
```js
const detectCompletion = async (item, event, opts) => {
  if (wasRenamedSuccessively(item, event)) {
    // We have a match, try to rebuild the incomplete event
    return rebuildIncompleteEvent(item, event, opts)
  } else if (itemDestinationWasDeleted(item, event)) {
    // We have a match, try to replace the incomplete event
    return buildDeletedFromRenamed(item, event)
  }
}
```

Then:

```js
// Look ma! No let ;)
const completion = await detectCompletion(item, event, opts)
if (!completion) continue
```

Then one could use `completion.rebuilt` to get the event, or maybe some `completion.ignored` prop?

*(also maybe we should find a way at some pointto make the code clear enough so we don't need all those comments, but this is out of the scope of this PR)*